### PR TITLE
Adding deletion param to emailAuth endpoint.

### PR DESCRIPTION
**Pull request by BenjaminSsempala**
#### Summary of Changes (What does this PR do?)

- Adjusting endpoint for reauthentication to cater for mobile In-app Account deletion

#### Status of maturity (all need to be checked before merging):

- [x] I've tested this locally
- [ ] I consider this code done
- [ ] This change ready to hit production in its current state
- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [Jira_card_number]() (If exists)

### CODE REVIEW

File: mobile/lib/services/rest_api.dart
Comment (Baalmart) [inline mobile/lib/services/rest_api.dart:226]: thanks @BenjaminSsempala . 

On a side note I am yet to prove, I am not very sure if PROTECTED auth-service endpoints can be used without the JWT header. The route definitions (code) in the microservice could shed more light on this.

I look forward to your analysis on this. 

Thanks!

Comment (BenjaminSsempala) [reply mobile/lib/services/rest_api.dart:226]: Thanks @Baalmart , 
The JWT header is also added to the post request right below this line.

Comment (Baalmart) [reply mobile/lib/services/rest_api.dart:226]: thanks @BenjaminSsempala

**Review by Baalmart — APPROVED** [review]

**Comment by Baalmart** [comment]
@NoahNsimbe , I will merge after your approval

**Review by NoahNsimbe — APPROVED** [review]
LGTM

**Comment by NoahNsimbe** [comment]
> @NoahNsimbe , I will merge after your approval

Done @Baalmart

**Comment by BenjaminSsempala** [comment]
> @BenjaminSsempala , this has conflicts, please resolve

Thanks @Baalmart I've resolved them

**Pull request merged**
